src/compiler/emitter.ts

```diff
@@ -60,6 +68,30 @@ module ts {
             sourceMaps: sourceMapDataList
         };
 
+        function isNodeDescendentOf(node: Node, ancestor: Node): boolean {
+            while (node) {
+                if (node === ancestor) return true;
+                node = node.parent;
+            }
+            return false;
+        }
+
+        function isUniqueLocalName(name: string, container: Node): boolean {
+            for (let node = container; isNodeDescendentOf(node, container); node = node.nextContainer) {

```

Comment:
If we decide that it is ok to always rename module parameter in module IIFE then we can get rid `isUniqueLocalName ` and all related stuff. Maybe it is not that bad given that this name is kind of implementation detail that does not affect external consumers of the module